### PR TITLE
Changed add_output API

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Changed add_output API

### CODE REVIEW

File: ngraph/core/src/function.cpp [hunk 1]
```diff
@@ -894,23 +893,23 @@ void ov::Function::add_output(const std::string& op_name, size_t output_idx) {
...
     for (const auto& input : port.get_target_inputs()) {
         // Do not add result if port is already connected with result
         if (ov::op::util::is_output(input.get_node())) {
-            return;
+            return input.get_node()->output(0);
         }
     }
     auto result = std::make_shared<ov::op::v0::Result>(port);
     add_results({result});
+    return result->output(0);
```
Comment (jane-intel) [inline ngraph/core/src/function.cpp:914]: I'm a bit confused. Why do we return the result output?

Comment (ilyachur) [reply ngraph/core/src/function.cpp:914]: Because it is equal to input.

**Comment by akuporos** [comment]
Could you please also add python tests here: https://github.com/openvinotoolkit/openvino/blob/master/runtime/bindings/python/tests/test_inference_engine/test_function.py

Their logic is quite close to cpp tests for `add_output`

**Comment by ilyachur** [comment]
All tests were added

**Comment by ilyachur** [comment]
@akuporos Could you take a look to python tests?

**Review by akuporos — APPROVED** [review]
Python LGTM

**Comment by ilyachur** [comment]
> Python side LGTM

@akuporos Python tests are failing. Could you please take a look why?

File: runtime/bindings/python/src/pyopenvino/graph/function.cpp [hunk 2]
```diff
@@ -589,6 +589,7 @@ void regclass_graph_Function(py::module m) {
         "add_outputs",
         [](ov::Function& self, py::handle& outputs) {
             int i = 0;
+            std::vector<ov::Output<ov::Node>> new_outputs;
```
Comment (akuporos) [inline runtime/bindings/python/src/pyopenvino/graph/function.cpp:592]: looks like it should be `ov::Output<const ov::Node>`, that should fix python tests

**Pull request merged**